wodles/aws/subscribers/s3_log_handler.py

```diff
@@ -305,3 +305,101 @@ def process_file(self, message_body: dict) -> None:
         for event in events_in_file:
             self.send_msg(event, dump_json=False)
         aws_tools.debug(f'{len(events_in_file)} events sent to Analysisd', 2)
+
+
+class AWSSecurityHubSubscriberBucket(AWSSubscriberBucket):
+    """Class for processing events from AWS S3 buckets containing Security Hub related logs."""
+
+    @staticmethod
+    def _add_event_type_fields(details: dict, event: dict):
+        """
+        Add the corresponding fields into the event if the details contain them.
+
+        Parameters
+        ----------
+        details : dict
+            Source dictionary containing the events from the log file.
+        event : dict
+            Destination dictionary to be added to the event sent to Wazuh.
+        """
+        key_actions = {
+            'findings': lambda source, destination: destination.setdefault('finding', source['findings'][0]),
+            'actionName': lambda source, destination: destination.setdefault('actionName', source['actionName']),
+            'actionDescription': lambda source, destination: destination.setdefault('actionDescription',
+                                                                                    source['actionDescription']),
+            'insightName': lambda source, destination: destination.setdefault('insightName', source['insightName']),
+            'insightArn': lambda source, destination: destination.setdefault('insightArn', source['insightArn']),
+            'resultType': lambda source, destination: destination.setdefault('resultType', source['resultType']),
+            'insightResults': lambda source, destination: destination.setdefault('insightResults',
+                                                                                 source['insightResults'])
+        }
+
+        for key, action in key_actions.items():
+            if key in details:
+                action(details, event)
```

Comment:
Related to the comment above, we could pass the key to the lambda as well to avoid repeating the strings multiple times

```suggestion
                action(details, event, key)
```